contracts/zero-ex/tests/MetaTransactionV2Test.t.sol

```diff
@@ -0,0 +1,311 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+  Copyright 2023 ZeroEx Intl.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "./utils/BaseTest.sol";
+import "forge-std/Test.sol";
+import "./utils/DeployZeroEx.sol";
+import "./utils/TestUtils.sol";
+import "../contracts/src/features/MetaTransactionsFeatureV2.sol";
+import "../contracts/src/features/interfaces/IMetaTransactionsFeatureV2.sol";
+import "../contracts/test/TestMintTokenERC20Transformer.sol";
+import "../contracts/src/features/libs/LibSignature.sol";
+import "src/features/libs/LibNativeOrder.sol";
+import "../contracts/test/tokens/TestMintableERC20Token.sol";
+import "@0x/contracts-utils/contracts/src/v06/errors/LibRichErrorsV06.sol";
+import "@0x/contracts-erc20/src/IEtherToken.sol";
+
+contract MetaTransactionTest is BaseTest, TestUtils {
+    DeployZeroEx.ZeroExDeployed zeroExDeployed;
+    address private constant ZERO_ADDRESS = 0x0000000000000000000000000000000000000000;
+    address private constant USER_ADDRESS = 0x6dc3a54FeAE57B65d185A7B159c5d3FA7fD7FD0F;
+    uint256 private constant USER_KEY = 0x1fc1630343b31e60b7a197a53149ca571ed9d9791e2833337bbd8110c30710ec;

```

Comment:
While `USER_KEY` and address constants were useful when thrown together for that [reference script](https://gist.github.com/abls/bb91ef9bb1ee1dd04dade9b2afeb2bf0#file-gas-s-sol-L22), we should probably have a better system for this in the test utils. A lot of things from that script are not good practice because that script was intended for a single use within an entirely different context.

Regarding improving our utils, a single `getSigner` function that can return only one signer isn't enough. It is common in our tests that we need multiple signers. Also, maybe we should be able to sign for any of the `account`s in `BaseTest`. Unsure exactly the best solution, but imo all the people writing tests should have a discussion about organization of the shared test utils at some point.